### PR TITLE
Avoid query results cache for very recent queries

### PR DESCRIPTION
**Pull request by tomwilkie**
@jtlisi  I've been away for last few days so don't know the background on this one.  I believe we were seeing a frequent query for the last ~5mins of data that was _very_ large (ie 30MB) in cache, and taking a few seconds to decode the JSON.   Can you confirm?

This change skips the cache lookup for very recent queries.

### CODE REVIEW

File: pkg/querier/frontend/results_cache.go [hunk 1]
```diff
@@ -86,6 +91,18 @@ func (s resultsCache) Do(ctx context.Context, r *queryRangeRequest) (*apiRespons
...
+	span, ctx := opentracing.StartSpanFromContext(ctx, "handle_fresh")
+	defer span.Finish()
+
+	response, err := s.next.Do(ctx, r)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return response, nil, nil
+}
```
Comment (tomwilkie) [inline pkg/querier/frontend/results_cache.go]: I think this whole function could be reduced to `return s.next.Do(ctx, r)` - in which case, we should just inline it in `resultsCache.Do` and use the return to un-indent the following block.

**Comment by gouthamve** [comment]
> I believe we were seeing a frequent query for the last ~5mins of data that was very large (ie 30MB) in cache, and taking a few seconds to decode the JSON

Yes. 


LGTM otherwise!

**Comment by bboreham** [comment]
The code is kinda similar to https://github.com/cortexproject/cortex/blob/master/pkg/chunk/chunk_store.go#L209-L213 - could they work off the same parameter?

**Comment by tomwilkie** [comment]
Could do, but they are applied to different components (query-frontend vs querier).  Probably not similar enough to warrant the re-organisation of the config TBH.

**Comment by jtlisi** [comment]
This code helped avoid the cache for a short duration query with a super high number of samples/carnality. In general though it will help avoid the frontend results cache for any query that has a duration too fresh to be present in the cache.

**Pull request merged**